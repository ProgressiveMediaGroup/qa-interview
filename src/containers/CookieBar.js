// Node_Modules
import React, { Component } from 'react'
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'
import PropTypes from 'prop-types'
import * as actions from '../actions'

// Styles
import '../styles/cookiebar.css'

export class CookieBar extends Component {
  render () {
    if (this.props.cookiebar.accept) {
      return null
    } else {
      return (
        <div className='cookie-bar'>
          <div>
            <span>This website might use cookies. Please let us know you agree to this:</span>

            <button onClick={this.props.actions.acceptCookies}>Accept Cookies</button>
          </div>
        </div>
      )
    }
  }
}

CookieBar.defaultProps = {
  cookiebar: {
    accept: false
  },
  actions: {
    acceptCookies: function () {}
  }
}

CookieBar.propTypes = {
  cookiebar: PropTypes.object,
  actions: PropTypes.object
}

function mapStateToProps (state) {
  return {
    cookiebar: state.cookiebar
  }
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CookieBar)
