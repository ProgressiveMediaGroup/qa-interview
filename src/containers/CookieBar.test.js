/* global it, expect, describe, jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import { CookieBar } from './CookieBar'

it('renders without crashing', () => {
  shallow(<CookieBar />)
})

it('should match snapshot', () => {
  const wrapper = shallow(<CookieBar />)
  expect(wrapper).toMatchSnapshot()
})

it('should show correctly if accept=false', () => {
  const cookiebar = {accept: false}
  const wrapper = mount(<CookieBar cookiebar={cookiebar} />)
  expect(wrapper.children().length).toBe(1)
})

it('should show correctly if accept=true', () => {
  const cookiebar = {accept: true}
  const wrapper = mount(<CookieBar cookiebar={cookiebar} />)
  expect(wrapper.children().length).toBe(0)
})

describe('Testing text', function () {
  it('should show correctly ', () => {
    const wrapper = mount(<CookieBar />)
    expect(wrapper.find('span').length).toBe(1)
  })

  it('should display correctly ', () => {
    const wrapper = mount(<CookieBar />)
    expect(wrapper.find('span').text()).toBe('This website might use cookies. Please let us know you agree to this:')
  })
})

describe('Testing button', function () {
  it('should show correctly ', () => {
    const wrapper = mount(<CookieBar />)
    expect(wrapper.find('button').length).toBe(1)
  })

  it('should display correctly ', () => {
    const wrapper = mount(<CookieBar />)
    expect(wrapper.find('button').text()).toBe('Accept Cookies')
  })
})

describe('Testing button functionality', function () {
  it('Test no defined onclick', () => {
    const wrapper = mount(
      <CookieBar />
    )
    expect(typeof wrapper.prop('actions').acceptCookies).toBe('function')
  })
  it('Test accepts actions prop', () => {
    const onclickMock = jest.fn()
    const actions = {acceptCookies: onclickMock}
    const wrapper = mount(
      <CookieBar actions={actions} />
    )
    expect(wrapper.prop('actions')).toBe(actions)
  })
  it('Test actions prop mocks correctly', () => {
    const onclickMock = jest.fn()
    const actions = {acceptCookies: onclickMock}
    const wrapper = mount(
      <CookieBar actions={actions} />
    )
    wrapper.prop('actions').acceptCookies()
    expect(onclickMock.mock.calls.length).toBe(1)
  })
  it('Test actions fires mocks correctly', () => {
    const onclickMock = jest.fn()
    const actions = {acceptCookies: onclickMock}
    const wrapper = mount(
      <CookieBar actions={actions} />
    )
    wrapper.find('button').simulate('click')
    expect(onclickMock.mock.calls.length).toBe(1)
  })
})
