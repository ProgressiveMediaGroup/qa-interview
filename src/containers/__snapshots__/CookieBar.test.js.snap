// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CookieBar
    actions={
        Object {
            "acceptCookies": [Function],
          }
    }
    cookiebar={
        Object {
            "accept": false,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <span>
                This website might use cookies. Please let us know you agree to this:
        </span>
        <button
                onClick={[Function]}
        >
                Accept Cookies
        </button>
</div>,
      "className": "cookie-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            This website might use cookies. Please let us know you agree to this:
</span>,
          <button
            onClick={[Function]}
>
            Accept Cookies
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This website might use cookies. Please let us know you agree to this:",
          },
          "ref": null,
          "rendered": "This website might use cookies. Please let us know you agree to this:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Accept Cookies",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Accept Cookies",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <span>
                    This website might use cookies. Please let us know you agree to this:
          </span>
          <button
                    onClick={[Function]}
          >
                    Accept Cookies
          </button>
</div>,
        "className": "cookie-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              This website might use cookies. Please let us know you agree to this:
</span>,
            <button
              onClick={[Function]}
>
              Accept Cookies
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This website might use cookies. Please let us know you agree to this:",
            },
            "ref": null,
            "rendered": "This website might use cookies. Please let us know you agree to this:",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Accept Cookies",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Accept Cookies",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
